### DISCUSSION TITLE
Add a warning message at the beginning of incoming mails

### DISCUSSION THREAD

**Discussion by WalterHof** [rspamd/rspamd#5175] [Q&A]
I am looking for a way to add a warning message at the beginning of incoming mails, e.g. “External mail - do not click on links!" I found this article: https://github.com/rspamd/rspamd/discussions/3679 but the warning is inserted at the end of the mail. 
Question: Has anyone already added a warning message at the beginning of every incoming mail and would they like to share their knowledge about this?

**Comment by syntaxys** [comment]
This should work:

/etc/rspamd/local.d/groups.conf
```
group "internal" {
    symbols = {
        "AUTHENTICATED" {
            description = "Internal email from authenticated user.";
        }
    }
}
```
/etc/rspamd/local.d/settings.conf
```
authenticated {
    priority = high;
    authenticated = yes;
    apply {
        symbols ["AUTHENTICATED"];
        }
    }
```
/etc/rspamd/local.d/force_actions.conf
```
rules {
    REWRITE_SUBJECT {
        action = "rewrite subject";
        expression = "!AUTHENTICATED";
        subject = ">>> Mail from Outa Space! <<< %s";
      }
      ... more rules ...
    }
```
